### PR TITLE
Build the service-idler

### PR DESCRIPTION
**Pull request by DirectXMan12**
This introduces image builds for
https://github.com/openshift/service-idler.

### CODE REVIEW

**Comment by DirectXMan12** [comment]
cc @sjenning re our conversation

File: projects/service-idler/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        uri: https://github.com/openshift/service-idler.git
+      type: Git
+      dockerfile: |
+				FROM openshift/origin-release:golang-1.10
+				COPY . /go/src/github.com/openshift/service-idler
+				RUN curl -L https://github.com/golang/dep/releases/download/v0.4.1/dep-linux-amd64 -o /go/bin/dep && chmod +x /go/bin/dep
+				RUN cd /go/src/github.com/openshift/service-idler && \
+						/go/bin/dep ensure -vendor-only
+				RUN cd /go/src/github.com/openshift/service-idler && \
+						CGO_ENABLED=0 go build -a -tags netgo -o service-idler-controller ./cmd/controller-manager/main.go
```
Comment (sjenning) [inline projects/service-idler/pipeline.yaml]: mixed tabs and spaces.  the highest of computer crimes! :P

Comment (DirectXMan12) [reply projects/service-idler/pipeline.yaml]: lol, I'll fix that

File: projects/service-idler/pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      to:
+        kind: ImageStreamTag
+        name: kube-state-metrics:latest
+    source:
+      dockerfile: |
+        FROM        centos:7
+        MAINTAINER  Solly Ross <sross@redhat.com>
+
+        COPY service-idler-controller /bin/service-idler-controller
+        VOLUME /tmp
```
Comment (kargakis) [inline projects/service-idler/pipeline.yaml]: Out of curiosity, why do you need a docker volume here?

Comment (DirectXMan12) [reply projects/service-idler/pipeline.yaml]: I don't think the Docker volume is necessary here.  I copy-paste-modified the Dockerfile from another one of these configurations, and forgot to remove that.  I'll do so.

**Comment by DirectXMan12** [comment]
gah, one of the references has the wrong name.  Issuing a follow-up PR...

**Pull request merged**